planning/behavior_path_planner/include/behavior_path_planner/scene_module/pull_over/pull_over_module.hpp

```diff
@@ -44,17 +47,27 @@ using nav_msgs::msg::OccupancyGrid;
 using visualization_msgs::msg::Marker;
 using visualization_msgs::msg::MarkerArray;
 
+using freespace_planning_algorithms::AbstractPlanningAlgorithm;
+using freespace_planning_algorithms::AstarParam;
+using freespace_planning_algorithms::AstarSearch;
+using freespace_planning_algorithms::PlannerCommonParam;
+using freespace_planning_algorithms::RRTStar;
+using freespace_planning_algorithms::RRTStarParam;
+
 enum PathType {
   NONE = 0,
   SHIFT,
   ARC_FORWARD,
   ARC_BACKWARD,
+  FREESPACE,
 };
 
 struct PUllOverStatus
 {
-  PullOverPath pull_over_path{};
+  std::shared_ptr<PullOverPath> pull_over_path{};
+  std::shared_ptr<PullOverPath> lane_parking_pull_over_path{};
   size_t current_path_idx{0};
+  bool require_increment_{true};
```

Comment:
I couldn't understand the purpose of this flag. Would you add an explanation?

Reply:
Added explanations and make the function in  https://github.com/autowarefoundation/autoware.universe/pull/2879/commits/3a74de4dd81529a074795f5c70a3a191d692fa58.